src/nni_manager/training_service/reusable/gpuScheduler.ts

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
```

Comment:
It seems the main logic of this scheduler is very similar with remote machine gpu scheduler,  suggest to refactor remote machine gpu scheduler and reuse it for `EnvironmentInformation`

Reply:
Let's minimum changes each time. The original GpuScheduler has many dependencies on remote machine schema, and new GpuScheduler changed schema of gpu info also.

The long term goal is to migrate remote machine to trialdispatcher. It can improve remote machine performance, and more stable by UT on trialDispatcher level.